docs/architecture/adrs/011-detector-fields.md

```diff
@@ -0,0 +1,143 @@
+# ADR 011: Detector fields
+
+This ADR documents the design and decisions for extending the detector API for various fields, to accomodate various outputs from models in the guardrails ecosystem.
+
+This serves as an extension to [ADR 003 - Detector API design](./003-detector-api-design.md). The detector API can be found [at this Github page](https://foundation-model-stack.github.io/fms-guardrails-orchestrator/?urls.primaryName=Detector+API).
+
+## Motivation
+
+Libraries like the [vllm-detector-adapter](https://github.com/foundation-model-stack/vllm-detector-adapter) provide the detector API and serve LLMs like [Granite Guardian](https://huggingface.co/collections/ibm-granite/granite-guardian-models-66db06b1202a56cf7b079562) and [Llama Guard](https://huggingface.co/meta-llama) as detectors easily thrugh [vLLM](https://github.com/vllm-project/vllm).
+
+As the models undergo development, more information is being provided on model output. Llama Guard will provide "unsafe" categories when input has been categorized as "unsafe" e.g. "unsafe\nS1" [ref](https://huggingface.co/meta-llama/Llama-Guard-3-11B-Vision), and Granite Guardian 3.2 began to provide additional information such as confidence e.g. `"No\n<confidence> High </confidence>` to indicate `No` risk and `High` confidence in that decision [ref](https://github.com/ibm-granite/granite-guardian/blob/main/cookbooks/granite-guardian-3.2/detailed_guide_vllm.ipynb).
+
+At the time of writing, the detector API endpoints return two types of responses:
+1. [`/text/contents`](https://foundation-model-stack.github.io/fms-guardrails-orchestrator/?urls.primaryName=Detector+API#/Text/text_content_analysis_unary_handler) returns a list of lists of detections with spans. Each list of detections corresponds to the respective content in `contents` provided in the user request, so `contents` with 2 texts would return a list of 2 lists.
+2. Other endpoints return a list of detections, without spans.
+
+Any placement of additional detector fields should account for the two types of responses.
+
+## Assumptions
+- This ADR considers fields that apply on each particular detection or decision made by the detector model.
+- Knowledge of future model plans is being restricted here, so only a few examples are given with already released model functionalities.
+
+## Decisions
+- We will add a new high-level field of `metadata` to account for additional model information from detectors. This field will provide a dictionary with string keys and arbitrary values, so that values are not constrained to particular types like strings or floats. This will enable flexibility and is how APIs like [Llama Stack](https://github.com/meta-llama/llama-stack) provides additional information, whether on datasets or models.
```

Comment:
nit: 
```suggestion
- We will add a new high-level field of `metadata` to account for additional information from detector models. This field will provide a dictionary with string keys and arbitrary values, so that values are not constrained to particular types like strings or floats. This will enable flexibility and is how APIs like [Llama Stack](https://github.com/meta-llama/llama-stack) provides additional information, whether on datasets or models.
```

Since its not really additional model information, but information about the output that model is providing.